Make it work on macOS and update to QT 6.x

moreaki opened the pull request.

gnudles commented on glmodelview.cpp:
```diff
@@ -24,31 +30,32 @@ GlModelView::GlModelView(QWidget *parent) :
-            program.link();
-            program.bind();
+            QOpenGLShaderProgram *program;
+            program = new QOpenGLShaderProgram(this);
```
Where is this memory allocation getting released?

moreaki replied:
Good question ;). I'm a bit swamped with work, but I'll try to update the PR.

gnudles committed:
Merge pull request #7 from gnudles/mac_integration

Mac integration

gnudles committed:
Now adapted to Qt6.

Squashed commit of the following:

commit e6d8d53ea9c2fc12c34b89a7bbf0680602270fa9
Author: Kapandaria <dvoreader@gmail.com>
Date:   Thu Feb 27 10:14:39 2025 +0200

    Update stereomaker.cpp

    Scaling with keep aspect ratio probably made the height as zero.

commit 94a112353e457068c946ef78507b55a4a2ff4a27
Author: Kapandaria <dvoreader@gmail.com>
Date:   Sun Feb 16 14:49:03 2025 +0200

    Update stereomaker.cpp

commit 5368dd7d335f8b80dbc2e17e5ca6bc9d467f7a6c
Author: Kapandaria <dvoreader@gmail.com>
Date:   Sun Feb 16 14:01:45 2025 +0200

    Update stereomaker.cpp

commit 169ad0276cdbd6e8c6a22616475c5071dcd0df38
Author: Kapandaria <dvoreader@gmail.com>
Date:   Sun Feb 16 13:55:40 2025 +0200

    Update stereomaker.h
